Tests/FunctionalTests/NetworkResponseHandlerFunctionalTests.swift

```diff
@@ -521,17 +531,25 @@ class NetworkResponseHandlerFunctionalTests: TestBase, AnyCodableAsserts {
 
         let dispatchEvents = getDispatchedEventsWith(type: TestConstants.EventType.EDGE, source: TestConstants.EventSource.RESPONSE_CONTENT)
         XCTAssertEqual(1, dispatchEvents.count)
-        guard let receivedData = dispatchEvents[0].data else {
-            XCTFail("Invalid event data")
-            return
+
+        let expected = """
+        {
+          "payload": [
+            {
+              "key": "s_ecid",
+              "maxAge": 15552000,
+              "value": "MCMID|29068398647607325310376254630528178721"
+            }
+          ],
+          "requestId": "123",
+          "type": "STRING_TYPE"
         }
+        """
 
-        let flattenReceivedData: [String: Any] = flattenDictionary(dict: receivedData)
-        XCTAssertEqual(5, flattenReceivedData.count)
-        XCTAssertEqual("123", flattenReceivedData["requestId"] as? String)
-        XCTAssertEqual("s_ecid", flattenReceivedData["payload[0].key"] as? String)
-        XCTAssertEqual("MCMID|29068398647607325310376254630528178721", flattenReceivedData["payload[0].value"] as? String)
-        XCTAssertEqual(15552000, flattenReceivedData["payload[0].maxAge"] as? Int)
+        assertExactMatch(
+            expected: expected,
+            actual: dispatchEvents[0],
+            pathOptions: CollectionEqualCount(scope: .subtree), ValueTypeMatch(paths: "type"))
```

Comment:
If I'm reading this correctly, this is testing that:

1. The number of entries in actual matches expected
2. The field "type" has the expected String type
3. Outside of field "type", the values of actual exactly match the values in expected.

If that's true then this is a good way to validate this test.